label_studio/storage/redis.py

```diff
@@ -0,0 +1,154 @@
+import json
+import os
+import logging
+import redis
+
+from .base import BaseStorage, BaseForm, StringField
+
+
+logger = logging.getLogger(__name__)
+
+def get_redis_connection(db = None):
+    if not db:
+        # This should never happen, but better to check than to accidentally 
+        # overwrite an existing database by choosing a wrong default:
+        logger.error("No redis db id passed!")
+    # Since tasks are always text, we use StrictRedis with utf-8 decoding.
+    return redis.StrictRedis(db=db, charset="utf-8", decode_responses=True)
+
+
+class RedisStorageForm(BaseForm):
+    path = StringField('Path', description='Storage prefix (optional)')
+    bound_params = dict(path='path')
+
+
+class RedisStorage(BaseStorage):
+    """Storage class for a redis database connection. 
+    
+    Inherits from BaseStorage.
+
+    Conventions:
+    A redis server hosts multiple, enumerated databases (0,1,2...).
+    The database used for storage should ONLY contain tasks. Any data already
+    in there might be lost / overwritten at any point in time and / or break
+    the functionality.
+
+    All keys are integers, but redis saves them as strings, so we sometimes
+    need to convert. The redis key is our prefix (e.g. "my_project/") + the 
+    labelstudio key as string (e.g. "3").
+
+    Example: If path is defined as "my_project", calling set(1, my_val) will
+    write to 
+    my_project/1 (the "/" is added if not already in path)
+    
+    Values are python dicts encoded as utf-8 strings.
+    """
+
+    description = 'Redis database connection'
+    
+    form = RedisStorageForm
+
+    def __init__(self, project_path, path=None, db=1, **kwargs):
+        """Initialize the storage.
+
+        If no path is provided manually (None or empty string), the 
+        project_path is instead used.
+
+        By default, db 1 is used for tasks.
+
+        Args:
+            project_path (str): Labelstudio project path
+            path (str or None, optional): The path that is used as a prefix. 
+                                          Defaults to None.
+            db (int): The Redis database to use.
+        """        
+        
+        # Check if a path was manually provided as an input parameter:
+        if not path or len(path) == 0: 
+            path = project_path
+
+        # Append "/" to path if necessary:
+        path = path.rstrip("/")+"/"
+        
+        super().__init__(path=path, project_path=project_path, **kwargs)
+
+        # Get redis database object and test connection:
+        self.r = get_redis_connection(db=db)
+        if not self.r.ping():
```

Comment:
Does it raise `redis.exceptions.ConnectionError` ?

Reply:
And anyway, without the Redis server enabled we can't proceed to work with `self.r` variable further - perhaps it's better to leave an exception here in order to prevent using this type of storage at an early stage

Reply:
Good point, I will modify this to `raise redis.exceptions.ConnectionError("No redis db id passed!")`

Reply:
Upon further consideration, I decided to instead raise ValueError if no db is provided and redis.exceptions.ConnectionError if the Ping fails.